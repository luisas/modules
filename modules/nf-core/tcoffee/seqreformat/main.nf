process TCOFFEE_SEQREFORMAT {
    tag "$meta.id"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'oras://community.wave.seqera.io/library/t-coffee_pigz:91ac7e26b23bb246':
        'community.wave.seqera.io/library/t-coffee_pigz:7d1373a24f76afe6' }"

    input:
    tuple val(meta), path(infile)

    output:
    tuple val(meta), path("${prefix}.txt"), emit: formatted_file
    path "versions.yml" , emit: versions


    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    export TEMP='./'
    t_coffee -other_pg seq_reformat \
        -in ${infile} \
        $args \
        > "${prefix}.txt"

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        tcoffee: \$( t_coffee -version | awk '{gsub("Version_", ""); print \$3}')
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    # Otherwise, tcoffee will crash when calling its version
    export TEMP='./'
    touch "${prefix}.txt"

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        tcoffee: \$( t_coffee -version | awk '{gsub("Version_", ""); print \$3}')
    END_VERSIONS
    """
}
