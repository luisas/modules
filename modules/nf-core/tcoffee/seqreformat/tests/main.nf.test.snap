{
    "seatoxin - stub": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                "versions.yml:md5,a9310e6837f0e4ddf6a5a6e8b465351f"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-19T08:52:48.068857066"
    },
    "seatoxin": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.txt:md5,fcd4691daf120c88ec5def7ac06fb562"
                ]
            ],
            [
                "versions.yml:md5,a9310e6837f0e4ddf6a5a6e8b465351f"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-19T08:52:29.963402675"
    }
}